builtin/method_dict.py

```diff
@@ -64,6 +66,89 @@ def Call(self, rd):
         return value.Null
 
 
+class Clear(vm._Callable):
+
+    def __init__(self):
+        # type: () -> None
+        pass
+
+    def Call(self, rd):
+        # type: (typed_args.Reader) -> value_t
+
+        dictionary = rd.PosDict()
+        rd.Done()
+
+        dictionary.clear()
+        return value.Null
+
+
+class Add(vm._Callable):
+
+    def __init__(self):
+        # type: () -> None
+        pass
+
+    def Call(self, rd):
+        # type: (typed_args.Reader) -> value_t
+
+        dictionary = rd.PosDict()
+        key = rd.PosStr()
+        inc_val = rd.OptionalValue()
+        rd.Done()
+
+        # If inc_val is provided, we treat it as the definitive type of the
+        # value to be incremented, if it isn't, we rely on the existing
+        # value in the dictionary (if that doesn't exist, the default is 0)
+        if inc_val is None:
+            right = value.Int(mops.ONE)  # type: value_t
```

Comment:
hm I'm on the fence about default 1 ... I think writing out 1 could be OK

```
call d->inc(k, 1)
call d->add(k, 1)
```

this seems harder to read:

```
call d->add(k)  # what are you adding?
```